gridcomps/History/Sampler/MAPL_EpochSwathMod.F90

```diff
@@ -193,6 +193,39 @@ subroutine config_accumulate (this, key, cf, rc)
   end subroutine config_accumulate
 
 
+  subroutine verify_epoch_equals_freq (this, frequency_from_list, swath_grid_label, rc)
+    class(samplerHQ)  :: this
+    integer, intent(in) :: frequency_from_list
+    character(len=*) , intent(in) :: swath_grid_label
+    integer, intent(out), optional :: rc
+    type(ESMF_Config) :: config_grid
+    integer :: hq_epoch_sec
+    integer :: freq_sec
+    integer :: local_swath_epoch_sec
+    integer :: time_integer
+    logical :: con
+    integer :: status
+
+    call ESMF_TimeIntervalGet(this%Frequency_epoch, s=hq_epoch_sec, _RC)
+    freq_sec = MAPL_nsecf( frequency_from_list )
+    config_grid = this%find_config( swath_grid_label )
+    call ESMF_ConfigGetAttribute(config_grid, value=time_integer, &
+         label=trim(swath_grid_label)//'.Epoch:', default=0, _RC)
+    local_swath_epoch_sec = MAPL_nsecf( time_integer )
+
+    con = (hq_epoch_sec == local_swath_epoch_sec) .AND. (hq_epoch_sec == freq_sec)
+    if (mapl_am_i_root()) then
+       if (.not. con) then
+          write(6, '(2x,a,2x,i10)') 'hq_epoch_sec', hq_epoch_sec
+          write(6, '(2x,a,2x,i10)') 'local_swath_epoch_sec', local_swath_epoch_sec
+          write(6, '(2x,a,2x,i10)') 'freq_sec', freq_sec
+       end if
+    end if
```

Comment:
Not in this pass, but please follow up with a PR to replace `write` with pFlogger calls.